source/common/grpc/async_client_impl.cc

```diff
@@ -184,8 +190,10 @@ void AsyncStreamImpl::initialize(bool buffer_body_for_retry) {
 void AsyncStreamImpl::onHeaders(Http::ResponseHeaderMapPtr&& headers, bool end_stream) {
   const auto http_response_status = Http::Utility::getResponseStatus(*headers);
   const auto grpc_status = Common::getGrpcStatus(*headers);
-  callbacks_.onReceiveInitialMetadata(end_stream ? Http::ResponseHeaderMapImpl::create()
-                                                 : std::move(headers));
+  if (!waiting_to_delete_on_remote_close_) {
```

Comment:
I am not following the if-statement here, are we saying if we half-close from the client side, we hides the initial metadata from GrpcServer from upper app layer ?